### PR TITLE
#166 Visualisation tool

### LINKED ISSUE #166: Enhancement + docs: visualisation tool for demos and to demo registration on README.md
# Issue description

It would be powerful to be able to visualise registration results achieved with the package. At the moment I think it would be extra nice if this is showcased on the README with a GIF of some sort to show updates of learning process visually for some trained network.

## Type of Issue

Please delete options that are not relevant.

- [x] New feature request
- [x] Documentation update

#### What's the expected result?
- Tool to plot results, maybe a keras style callback during training?
- A nice diagram/GIF/pictures of registration achieved using the package from a demo on the README to visually showcase and add some flair/impact.

### PR DESCRIPTION
**Pull request by s-sd**
# Description

Visualisation tool added with corresponding unit tests and docs. Currently it supports 4 functionalities:
- gif over image slices (of single or multiple images at a time)
- gif that shows warping (of single or multiple images at a time)
- a plot which combines different slices fromm different images (like the one we did for the edu challenge notebook)
- tiled gif of multiple images with an animation over the slices of each

Entrypoint for the CLI tool has also been added to the setup.py file. All functionalities are accessible either via the CLI tool called `deepreg_vis` or by importing python functions.

Please see the newly added doc for more details. 

Fixes #166 

Note: This PR is to merge to the develop branch.

## Type of change

What types of changes does your code introduce to DeepReg? _Put an `x` in the boxes that
apply_

- [x] New feature (non-breaking change which adds functionality)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR.
If you're unsure about any of them, don't hesitate to ask. We're here to help! This is
simply a reminder of what we are going to look for before merging your code._

- [x] I have
      [installed pre-commit](https://deepreg.readthedocs.io/en/latest/contributing/setup.html)
      and formatted all changed files.
- [x] My commits' message styles matches
      [our requested structure](https://deepreg.readthedocs.io/en/latest/contributing/commit.html),
      e.g. `Issue #<issue number>: detailed message`.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by YipengHu** [comment]
Nice! I will have a go with it later.

**Comment by s-sd** [comment]
@YipengHu, sorry I spotted a small typo in the doc. Corrected it now

**Comment by s-sd** [comment]
@YipengHu, also one small note: the `--save-path` or `-s` that you specify as the output of the visualisation must already exist. So, the `logs` directory must be created before running the examples in the doc.

File: deepreg/vis.py [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+
+            anim.save(path_to_anim_save)
+            print("Animation saved to:", path_to_anim_save)
+
+
+def gif_tile_slices(img_paths, save_path=None, size=(2, 2), fname=None, interval=50):
+    """
+    Creates tiled gif over slices of multiple images
+
+    :param img_paths : list or comma separated string of image paths
```
Comment (mathpluscode) [inline deepreg/vis.py:184]: I'm just wondering if `:param xxx :` works, there's a space before `:`

Comment (s-sd) [reply deepreg/vis.py:184]: Sorry, that's a typo. I will correct it now

**Comment by YipengHu** [comment]
@s-sd minor text updated, have a look see if correct. Otherwise happy to merge.

**Review by YipengHu — DISMISSED** [review]

**Comment by s-sd** [comment]
@YipengHu, sorry, codecov was blocking the merge because of a couple of lines missed in the unit test for vis.py so I had to make another commit to add the tests which cover those lines. It would be great if you could review/ approve now.

**Review by YipengHu — APPROVED** [review]

**Pull request merged**